src/transformers/models/fastspeech2_conformer/tokenization_fastspeech2_conformer.py

```diff
@@ -0,0 +1,176 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Team and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tokenization classes for FastSpeech2Conformer."""
+import json
+import os
+from typing import Optional, Tuple
+
+import regex
+
+from ...file_utils import requires_backends
+from ...tokenization_utils import PreTrainedTokenizer
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+VOCAB_FILES_NAMES = {"vocab_file": "vocab.json"}
+
+PRETRAINED_VOCAB_FILES_MAP = {
+    "vocab_file": {
+        "connor-henderson/fastspeech2_conformer": "https://huggingface.co/connor-henderson/fastspeech2_conformer/raw/main/vocab.json",
+    },
+}
+
+PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES = {
+    # Set to somewhat arbitrary large number as the model input
+    # isn't constrained by the relative positional encoding
+    "connor-henderson/fastspeech2_conformer": 4096,
+}
+
+
+class FastSpeech2ConformerTokenizer(PreTrainedTokenizer):
```

Comment:
LGTM - @ArthurZucker would you mind casting your eye over the tokenizer when you get the chance? Thanks!

Reply:
Sure!

Reply:
And more generally the PR when you get the chance, this is ready for a core maintainers' review :)